Notes/Algorithms/Data Structures/Hash-Tables.c

```diff
@@ -4,34 +4,231 @@
 
 #define SUCCESS 0
 #define ERROR -1
-#define OVERFLOW -2
-#define EMPTY -3
+
+#define TEST 9
+#define EXPECTED 10
+
+#define SIZE 10
 
 // Hash struct
-typedef struct Hash {
+typedef struct Item {
     int key;
     int value;
-} Hash;
+} Item;
 
-void print() {
+//create global testHash and expectedHash for testFunctions to use
+Item* testHash[SIZE];
+Item* expectedHash[SIZE];
 
+// traverse function is for visual confirmation of differences between hashTables when building functionality. 
+// void traverse() {
+//     for (int i = 0; i < SIZE; i++) {
+//         int value = testHash[i]->value;
+//         int key = testHash[i]->key;        
+//         printf("Test Key: %d Value: %d \n", key, value);
+//     }
+//     for (int i = 0; i < SIZE; i++) {
+//         int value = expectedHash[i]->value;
+//         int key = expectedHash[i]->key;
+//         printf("Expected Key: %d Value: %d \n", key, value);
+//     }
+// }
+
+//compare checks the testHash and expectedHash for parity
+// if the two HashTables match returns true
+// if a mismatch occurs returns false
+bool compare() {
+    for (int i = 0; i < SIZE; i++) {
+        int testKey = testHash[i]->key;
+        int testValue = testHash[i]->value;
+        int expectedKey = expectedHash[i]->key;
+        int expectedValue = expectedHash[i]->value;
+        if (testKey != expectedKey || testValue != expectedValue) {
+            return false;
+        }
+    }
+    return true;
 }
 
-void traverse() {
+// hashing takes in a key
+// returns a "hashed" index based on the SIZE of the hashTable
+int hashing(int key) {
+    return key % SIZE;
+}
 
+// creates a new Item struct
+// returns new struct
+Item * createItem(int key, int value, int success) {
+    Item * item = malloc(sizeof(Item));
+    if (item == NULL) {
+    } else {
+        item->key = key;
+        item->value = value;
+        success = SUCCESS;
+        return item;
+    }
 }
 
-bool compare() {
-    return true;
-    return false;
+// insert takes in a key and value
+// calls a helper function to create an Item struct with the provided key and value
+// adds the new Item to the hashTable
+// if the expected value is in the hashTable at the expected key then returns a success code
+void insert(int type, int key, int value, int success) {
```

Comment:
`success` is being passed in as a value, so changes to this variable will not be reflected outside the scope of this function. Might be better to just return the error/success value.

Reply:
Addressed to return success/error codes in [commit f4e9a6ebcdbe44e3b0fd1d3c202f73251917ae7a](https://github.com/kcarrel/Training-Notes/pull/12/commits/f4e9a6ebcdbe44e3b0fd1d3c202f73251917ae7a)